### DISCUSSION TITLE
Put a Running Version ID in Somewhere For Recognition/Debugging

### DISCUSSION THREAD

**Discussion by daveclifford** [DCC-EX/EX-T3-WiFi#71] [Ideas]
Alex,

It might be beneficial for you to put a Running Version ID whenever you update the code and make it evident to the user, so he/she can cite the Version ID in correspondence.

The only issue is that you have to change the ID number each time you improve/fix the code.

You could use the character string to show for 5 seconds **"Hayward Junction EX-T3-WiFi-R1. n"** when the Wi-Fi Throttle is first turned on.

It may help you in troubleshooting and get some good recognition for your effort!

Dave Clifford

**Comment by alex-code** [comment]
I'm working on some updates that will start using version numbers.
The plan is to then release versioned binary builds so you won't need to use VSCode and compile it yourself.

**Reply by daveclifford** [reply]
Geniuses think alike !! You never know where your work will lead to....

When I created a code release, there was always a version/date so that the user could give us something to tell us how old the code was and we could then tell them - UPGRADE - the problem was fixed 3-5 releases ago.... (haha).......

Dave

**Reply by alex-code** [reply]
There's still some more testing to do on building a single binary (need to merge the firmware and SPIFFS bins) then test the web uploading.
Bit more testing on settings upgrading too in case I reorganise the JSON structure 😉, this should allow the firmware to auto update from old to new settings without loosing them.

**Reply by daveclifford** [reply]
Kool!!!!
